src/mscorlib/src/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -36,57 +36,37 @@ private static EqualityComparer<T> CreateComparer()
             Contract.Ensures(Contract.Result<EqualityComparer<T>>() != null);
             
             object result = null;
-            RuntimeType t = (RuntimeType)typeof(T);
+            var type = (RuntimeType)typeof(T);
             
-            // Specialize type byte for performance reasons
-            if (t == typeof(byte)) {
+            // Specialize for byte so Array.IndexOf is faster.
+            if (type == typeof(byte))
```

Comment:
> Getting rid of IEquatable<T> is possible too but a bit harder. 

Actually, it should not be that hard - it can use the same pattern as what is used on the nulalble path: `typeof(IEquatable<>).MakeGenericType(u).IsAssignableFrom(u)`